gateway/api/views.py

```diff
@@ -436,9 +436,16 @@ def logs(self, request, pk=None):  # pylint: disable=invalid-name,unused-argumen
         tracer = trace.get_tracer("gateway.tracer")
         ctx = TraceContextTextMapPropagator().extract(carrier=request.headers)
         with tracer.start_as_current_span("gateway.job.logs", context=ctx):
-            job = self.get_object()
+            job = Job.objects.filter(pk=pk).first()
```

Comment:
I would add a `if job is None return Reponse(404)` if not it will break if job doesn't exist.